chia/consensus/block_record.py

```diff
@@ -117,3 +154,66 @@ def ip_iters(self, constants: ConsensusConstants) -> uint64:
 
     def sp_total_iters(self, constants: ConsensusConstants) -> uint128:
         return uint128(self.sp_sub_slot_total_iters(constants) + self.sp_iters(constants))
+
+    def to_block_record_db(self) -> BlockRecordDB:
```

Comment:
I think `BlockRecordDB` should be private in `block_store.py` and any conversion functions should also live in that file.
Keeping it private makes the surface area to the block store smaller and makes it easier to change its implementation without affecting the rest of the code base.

e.g. it opens the door for a v3 database format where we don't store all the duplicate fields in both the `BlockRecord` and `FullNode`